### ISSUE TITLE
KeyError raised by the env.render()

### ISSUE THREAD

**Issue by ashe1098** [minerllabs/minerl#365]
I tried the K-means exploration straight from [the tutorial](https://minerl.io/docs/tutorials/k-means.html).
```python
...
while not done:
    # Let's use a frame skip of 4 (could you do better than a hard-coded frame skip?)
    if i % 4 == 0:
        action = {
            'vector': kmeans.cluster_centers_[np.random.choice(NUM_CLUSTERS)]
        }

    obs, reward, done, info = env.step(action)
    env.render()
...
```
But the `env.render()` raises a KeyError and the program freezes.
```
DEBUG:minerl.env.malmo.instance.90e134:[13:17:19] [Server thread/INFO]: MineRLAgent joined the game
DEBUG:minerl.env.malmo.instance.90e134:[13:17:19] [Server thread/INFO]: [STDOUT]: SERVER enter state: BUILDING_WORLD
DEBUG:minerl.env.malmo.instance.90e134:[13:17:19] [Server thread/INFO]: [STDOUT]: SERVER request state: WAITING_FOR_AGENTS_TO_ASSEMBLE
DEBUG:minerl.env.malmo.instance.90e134:[13:17:19] [Server thread/INFO]: [STDOUT]: SERVER enter state: WAITING_FOR_AGENTS_TO_ASSEMBLE
DEBUG:minerl.env.malmo.instance.90e134:[13:17:19] [Server thread/INFO]: [STDOUT]: Experiment requires: 
DEBUG:minerl.env.malmo.instance.90e134:[13:17:19] [Server thread/INFO]: [STDOUT]: >>>> MineRLAgent
DEBUG:minerl.env.malmo.instance.90e134:[13:17:20] [Client thread/INFO]: [STDOUT]: ***Telling server we are ready - MineRLAgent
DEBUG:minerl.env.malmo.instance.90e134:[13:17:21] [Client thread/INFO]: [STDOUT]: CLIENT request state: RUNNING
DEBUG:minerl.env.malmo.instance.90e134:[13:17:21] [Client thread/INFO]: [STDOUT]: CLIENT enter state: RUNNING
DEBUG:minerl.env.malmo.instance.90e134:[13:17:21] [Server thread/INFO]: [STDOUT]: SERVER request state: RUNNING
DEBUG:minerl.env.malmo.instance.90e134:[13:17:21] [Server thread/INFO]: [STDOUT]: SERVER enter state: RUNNING
DEBUG:minerl.env.malmo.instance.90e134:[13:17:21] [Client thread/INFO]: [STDOUT]: Resized the window
DEBUG:minerl.env.malmo.instance.90e134:[13:17:21] [Client thread/INFO]: [STDOUT]: Unpausing
DEBUG:minerl.env.malmo.instance.90e134:[13:17:21] [Client thread/INFO]: [CHAT] §l400...
0it [00:00, ?it/s]Traceback (most recent call last):
  File "tutorial.py", line 88, in <module>
    kmean()
  File "tutorial.py", line 77, in kmean
    env.render()
  File "/Users/ashe/.local/share/virtualenvs/MineRL-APCPXVva/lib/python3.7/site-packages/gym/core.py", line 233, in render
    return self.env.render(mode, **kwargs)
  File "/Users/ashe/.local/share/virtualenvs/MineRL-APCPXVva/lib/python3.7/site-packages/minerl/env/core.py", line 688, in render
    self._renderObs(self._last_obs, self._last_ac)
  File "/Users/ashe/.local/share/virtualenvs/MineRL-APCPXVva/lib/python3.7/site-packages/minerl/env/core.py", line 681, in _renderObs
    self.viewer.render(obs, 0, 0, ac, 0, 1)
  File "/Users/ashe/.local/share/virtualenvs/MineRL-APCPXVva/lib/python3.7/site-packages/minerl/viewer/trajectory_display.py", line 149, in render
    self.process_observations(obs)
  File "/Users/ashe/.local/share/virtualenvs/MineRL-APCPXVva/lib/python3.7/site-packages/minerl/viewer/trajectory_display.py", line 356, in process_observations
    action_vec = obs['vector']
KeyError: 'vector'
DEBUG:minerl.env.malmo.instance.90e134:[13:17:24] [Server thread/INFO]: [STDOUT]: Warning: managed 4/20 ticks this second.
INFO:minerl.env.malmo:Minecraft process psutil.Process(pid=90135, status='terminated', exitcode=0, started='13:16:14') terminated with exit code 0
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/concurrent/futures/process.py", line 101, in _python_exit
    thread_wakeup.wakeup()
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/concurrent/futures/process.py", line 89, in wakeup
    self._writer.send_bytes(b"")
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/multiprocessing/connection.py", line 183, in send_bytes
    self._check_closed()
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/multiprocessing/connection.py", line 136, in _check_closed
    raise OSError("handle is closed")
OSError: handle is closed
Exception in thread QueueManagerThread:
Traceback (most recent call last):
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/concurrent/futures/process.py", line 368, in _queue_management_worker
    thread_wakeup.clear()
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/concurrent/futures/process.py", line 92, in clear
    while self._reader.poll():
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/multiprocessing/connection.py", line 255, in poll
    self._check_closed()
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/multiprocessing/connection.py", line 136, in _check_closed
    raise OSError("handle is closed")
OSError: handle is closed
```

I tried [no_op_test.py](https://github.com/minerllabs/minerl/blob/f338ff32ddd0cc736ab80924a3e1f5d1cba98edb/tests/excluded/no_op_test.py) and got the same KeyError.
But [simple_env_test.py](https://github.com/minerllabs/minerl/blob/3780a1b2ba36886cea500d92aa5374a585f96248/tests/excluded/simple_env_test.py) worked, so maybe only the feature of `render` for `vector` isn't working.

**Comment by MadcowD** [comment]
I think this is a pretty easy fix, apologies for how long it's taken to respond. I'll get right on this.

**Comment by MadcowD** [comment]
It's strange that trechop_vector_obf doesn't have a vector...

**Comment by ashe1098** [comment]
I checked the MineRLTreechopVectorObf and it definitely has a vector.
There seems to be a glitch in the way `env` is processing the data to perform `render()`.

**Comment by prabhasak** [comment]
Hello. Turns out the env.render() issue is common across all vector_obf environments. The vector env observations seem to be the same as the equivalent basic env observations. Printing [``self._last_obs``](https://github.com/minerllabs/minerl/blob/master/minerl/env/core.py#L688) for ``MineRLObtainDiamondVectorObf-v0`` gives:

```
OrderedDict([('equipped_items.mainhand.damage', array(0)), ('equipped_items.mainhand.maxDamage', array(0)), 
('equipped_items.mainhand.type', 'none'), ('inventory', OrderedDict([('coal', array(0)), ('cobblestone', array(0)), 
('crafting_table', array(0)), ('dirt', array(0)), ('furnace', array(0)), ('iron_axe', array(0)), ('iron_ingot', array(0)), 
('iron_ore', array(0)), ('iron_pickaxe', array(0)), ('log', array(0)), ('planks', array(0)), ('stick', array(0)), 
('stone', array(0)), ('stone_axe', array(0)), ('stone_pickaxe', array(0)), ('torch', array(0)), 
('wooden_axe', array(0)), ('wooden_pickaxe', array(0))])),
 ('pov', array([[[ 98, 129, 185],
        [ 98, 129, 186],
        [ 99, 130, 187],
        ...,
        [ 97, 127, 181],
        [ 97, 126, 180],
        [ 97, 126, 179]],

       [[ 98, 128, 184],
        [ 98, 129, 185],
        [ 98, 129, 186],
        ...,
        [ 97, 126, 180],
        [ 97, 126, 179],
        [ 96, 125, 178]],

       [[ 98, 128, 183],
        [ 98, 128, 184],
        [ 98, 129, 185],
        ...,
        [ 96, 125, 179],
        [ 96, 125, 178],
        [ 96, 125, 177]],

       ...,

       [[ 75,  98,  46],
        [ 75,  98,  47],
        [ 75,  98,  46],
        ...,
        [111, 106, 101],
        [111, 106, 101],
        [110, 105, 102]],

       [[ 67,  87,  41],
        [ 86, 111,  53],
        [111, 106, 101],
        [111, 106, 101]],
       [[ 86, 111,  53],
        [ 86, 112,  53],
        [111, 106, 101],
        [111, 106, 101],
        [111, 106, 101]]], dtype=uint8))])
```

which is the same output for the equivalent basic env ``MineRLObtainDiamond-v0``.

**Comment by MadcowD** [comment]
Oh!!! I know exactly how to fix this. Thanks for the investigation ~~~ been working on other features like Multiagent etc.

**Comment by MadcowD** [comment]
This will be released in the newest version of MineRL though you can just install from source and it should work.

**Comment by prabhasak** [comment]
FYI this is still not fixed in the latest 0.3.6 version.
